File: test/react-native/features/fixtures/reactnative/scenarios/MetadataNativeScenario.kt
package com.reactnative.scenarios

import android.content.Context
import com.bugsnag.android.Bugsnag

class MetadataNativeScenario(context: Context): Scenario(context) {

    override fun run() {
        super.run()
        Bugsnag.addMetadata("nativedata", "some_more_data", "set via client")
        Bugsnag.notify(generateException(), { event ->
            event.addMetadata("nativedata", "even_more_data", "set via event")
            true
        })
    }
}

File: test/react-native/features/fixtures/ios-module/Scenarios/MetadataNativeScenario.m
#import "MetadataNativeScenario.h"

@implementation MetadataNativeScenario

- (void)run: (RCTPromiseResolveBlock)resolve
     reject:(RCTPromiseRejectBlock)reject {
  NSException *exception = [[NSException alloc] initWithName:@"NSException" reason:@"MetadataNativeScenario" userInfo:nil];
  [Bugsnag addMetadata:@"set via client" withKey:@"some_more_data" toSection:@"nativedata"];
  [Bugsnag notify:exception block:^BOOL(BugsnagEvent *event) {
    [event addMetadata:@"set via event" withKey:@"even_more_data" toSection:@"nativedata"];
    return YES;
  }];

}

@end
